### PR TITLE
Improve concurrency headers documentation

### PR DESCRIPTION
**Pull request by leplatrem**
capture d ecran de 2016-03-07 15-12-00

@n1k0 r?

### CODE REVIEW

File: cliquet_docs/api/timestamps.rst [hunk 1]
```diff
@@ -56,6 +56,14 @@ In order to check that the client version has not changed, a ``If-None-Match``
 request header can be used. If the response is ``304 Not Modified`` then
 the cached version if still good.
```
Comment (n1k0) [inline cliquet_docs/api/timestamps.rst]: Nit: *is* still good.

File: cliquet_docs/api/resource.rst [hunk 2]
```diff
@@ -167,7 +167,7 @@ When filtering on ``last_modified`` every deleted records will appear in the
 list with a ``deleted`` flag and a ``last_modified`` value that corresponds
 to the deletion event.
 
-If the request header ``If-None-Match`` is provided as described in
+If the request header ``If-None-Match: "<timestamp>"`` is provided as described in
```
Comment (n1k0) [inline cliquet_docs/api/resource.rst]: Nit: if the If-None-Match header is provided

File: cliquet_docs/api/resource.rst [hunk 3]
```diff
@@ -355,9 +355,14 @@ The POST response body is a JSON mapping containing:
 - ``permissions``: *optional* a json dict containing the permissions for
   the requested resource.
 
-If the request header ``If-Match`` is provided, and if the record has
+If the request header ``If-Match: "<timestamp>"`` is provided as described in
+the :ref:`section about timestamps <server-timestamps>`, and if the collection has
 changed meanwhile, a ``412 Precondition failed`` error is returned.
```
Comment (n1k0) [inline cliquet_docs/api/resource.rst]: We should document the `details` property sent along the JSON body, useful to know about the conflicting record data.

File: cliquet_docs/api/resource.rst [hunk 4]
```diff
@@ -440,8 +445,7 @@ take the value of the created record but is bumped into the future as usual.
 HTTP Status Codes
 -----------------
 
-.. * ``200 OK``: This record already exists, here is the one stored on the database;
-
+* ``200 OK``: This record already exists, the one stored on the database is returned.
```
Comment (n1k0) [inline cliquet_docs/api/resource.rst]: Nit: stored *in* the database

File: cliquet_docs/api/timestamps.rst [hunk 5]
```diff
@@ -64,9 +72,32 @@ In order to prevent race conditions, like overwriting changes occured in the int
...
+
+Concurrency control also allows to make sure a creation won't overwrite any record:
+
++---------------------+-----------------------------+------------+
+| **If-None-Match: ***| Id exists                   | Id unknown |
++=====================+=============================+============+
+| POST {resource}     | ``412 Precondition failed`` | Create     |
++---------------------+-----------------------------+------------+
+| PUT {resource}/{id} | ``412 Precondition failed`` | Create     |
++---------------------+-----------------------------+------------+
```
Comment (n1k0) [inline cliquet_docs/api/timestamps.rst]: These tables would be a little more obvious (and helpful) to me if they featured the different operations with different results in a same table. Right now you have to compare two distinct tables in the page, which is possibly confusing to the reader.

**Comment by leplatrem** [comment]
capture d ecran de 2016-03-10 11-01-23

**Comment by leplatrem** [comment]
@n1k0 how does it look now?

**Comment by n1k0** [comment]
Much MUCH better, kudos! r+ from me.

**Comment by leplatrem** [comment]
Thanks!

**Pull request merged**